codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -129,79 +202,74 @@ public boolean contains(T name) {
     public boolean containsObject(T name, Object value) {
         return contains(name, valueConverter.convertObject(checkNotNull(value, "value")));
     }
 
     @Override
     public boolean containsBoolean(T name, boolean value) {
-        return contains(name, valueConverter.convertBoolean(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertBoolean(value));
     }
 
     @Override
     public boolean containsByte(T name, byte value) {
-        return contains(name, valueConverter.convertByte(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertByte(value));
     }
 
     @Override
     public boolean containsChar(T name, char value) {
-        return contains(name, valueConverter.convertChar(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertChar(value));
     }
 
     @Override
     public boolean containsShort(T name, short value) {
-        return contains(name, valueConverter.convertShort(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertShort(value));
     }
 
     @Override
     public boolean containsInt(T name, int value) {
-        return contains(name, valueConverter.convertInt(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertInt(value));
     }
 
     @Override
     public boolean containsLong(T name, long value) {
-        return contains(name, valueConverter.convertLong(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertLong(value));
     }
 
     @Override
     public boolean containsFloat(T name, float value) {
-        return contains(name, valueConverter.convertFloat(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertFloat(value));
     }
 
     @Override
     public boolean containsDouble(T name, double value) {
-        return contains(name, valueConverter.convertDouble(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertDouble(value));
     }
 
     @Override
     public boolean containsTimeMillis(T name, long value) {
-        return contains(name, valueConverter.convertTimeMillis(checkNotNull(value, "value")));
+        return contains(name, valueConverter.convertTimeMillis(value));
     }
 
-    @Override
     @SuppressWarnings("unchecked")
-    public boolean contains(T name, T value, Comparator<? super T> valueComparator) {
-        Object values = map.get(name);
-        if (isList(values)) {
-            List<T> values0 = (List<T>) values;
-            for (int i = 0; i < values0.size(); i++) {
-                if (valueComparator.compare(value, values0.get(i)) == 0) {
-                    return true;
-                }
+    @Override
+    public boolean contains(T name, T value) {
+        return contains(name, value, JAVA_HASHER);
+    }
+
+    protected boolean contains(T name, T value, HashingStrategy<? super T> valueHashingStrategy) {
```

Comment:
Is this method meant to be overridden or accessed by subclasses? How about making it final (and also private if possible)?